File: 821 - Shorteset Distance to a Character.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int[] shortestToChar(String s, char c) {
        Set<Integer> cset = new HashSet<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c) {
                cset.add(i);
            }
        }

        int[] res = new int[s.length()];
        int dist, minDist;
        for (int i = 0; i < s.length(); i++) {
            minDist = s.length();
            for (int index : cset) {
                dist = Math.abs(index - i);
                minDist = minDist < dist ? minDist : dist;
            }
            res[i] = minDist;
        }
        return res;
    }
}

File: 821 - Shorteset Distance to a Character.cpp
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    vector<int> shortestToChar(string s, char c) {
        vector<int> res(s.size());
        int prev = -s.size();
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == c) {
                prev = i;
            }
            res[i] = i - prev;
        }

        prev = s.size() * 2;
        for (int i = s.size() - 1; i >= 0; i--) {
            if (s[i] == c) {
                prev = i;
            }
            res[i] = res[i] < prev - i ? res[i] : prev - i;
        }
        return res;
    }
};